### ISSUE TITLE
CLI codegen fails with spaces in project name

### ISSUE THREAD

**Issue by justinormont** [dotnet/machinelearning#3792]
The AutoML CodeGen can't handle spaces in the project name, or it produces a corrupt .cs file.

Repro:
`mlnet auto-train --dataset analcatdata_germangss.csv  --ml-task multiclass-classification --label-column-name "Political_system" --max-exploration-time 60 --name "Analyzing Categorical Data" --output-path /tmp/blah/`

Note the spaces in the project name: 
`--name "Analyzing Categorical Data" `

Dataset: 
https://www.openml.org/data/get_csv/53559/analcatdata_germangss.arff

This produces corrupted .cs files with replicas of the project name.

For example, `Program.cs`:
```c#
using System;
using System.IO;
using System.Linq;
using Microsoft.ML;
using Analyzing Categorical Data.Model.DataModels;

namespace Analyzing Categorical Data.ConsoleApp
{
    class Program
{
```

You'll note `Analyzing Categorical Data` is replicated into `Program.cs` with the spaces intact. We need to sanitize the project name to fit namespace/variable name requirements.

Currently we use a simple sanitization on the column names:
https://github.com/dotnet/machinelearning/blob/fac974daaec029b732802c1d0802818a27ab37be/src/mlnet/Utilities/Utils.cs#L47-L50

We may be able to reuse that sanitizer.

**Comment by justinormont** [comment]
There's a related issue with internationalization.

Using a project name of "학습유형들에", the CodeGen project compiles though fails on execution.

Error:
`Unhandled Exception: System.IO.FileNotFoundException: Could not load file or assembly '학습유형들에.Model, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.`

**Comment by zHaytam** [comment]
Hello, is this issue still open? I would like to take it as my first issue.

**Comment by justinormont** [comment]
@zHaytam: Yes this is still open. You can verify by calling the `mlnet` CLI command with a project name with a space in it, as given above, `--name "Analyzing Categorical Data"`. This causes the generated project to have an invalid class name error when the generated code is compiled.

You're welcome to take on the task. 

Currently the code for the CLI is in the [AutoML feature branch](https://github.com/dotnet/machinelearning/tree/features/automl), so your modification will flow into that feature branch.

Welcome to the project!

**Comment by zHaytam** [comment]
I can't seem to find the project in the solution, what's its name please?

**Comment by zHaytam** [comment]
I have a couple of questions:

1. Wouldn't it better for `Analyzing Categorical Data` to become `Analyzing.Categorical.Data` instead of `Analyzing_Categorical_Data`?
2. If we use `Sanitize` on names that aren't even english (like your example `학습유형들에`), the namespace will just become `______.ConsoleApp`, which is a valid namespace but not a great one, should we still do it?

**Comment by justinormont** [comment]
The sanitization for C# needs to produce valid namespaces and variable names.

As long as it meets the criteria for a C# namespace & variable name we can expand the sanitization to include internationalized character sets. We should also check that internationalization is currently not handled since a quick reading of [Char.IsLetterOrDigit](https://docs.microsoft.com/en-us/dotnet/api/system.char.isletterordigit?view=netcore-3.0) looks like it should support unicode.

> Should the names of the csproj and sln files change too?

Yes, we should likely apply the sanitization across the board.

**Comment by zHaytam** [comment]
I have created a PR for this but can't seem to have the permission to request reviewers, not sure what to do next :/
